src/main/scala/org/apache/spark/sql/redis/DefaultSource.scala

```diff
@@ -8,25 +8,29 @@ class DefaultSource extends RelationProvider
   //  with SchemaRelationProvider
   with CreatableRelationProvider {
 
-  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]): BaseRelation = {
+  override def createRelation(sqlContext: SQLContext,
+                              parameters: Map[String, String]): BaseRelation = {
     new RedisSourceRelation(sqlContext, parameters, userSpecifiedSchema = None)
   }
 
   /**
     * Creates a new relation by saving the data to Redis
     */
-  override def createRelation(sqlContext: SQLContext, mode: SaveMode, parameters: Map[String, String], data: DataFrame): BaseRelation = {
+  override def createRelation(sqlContext: SQLContext, mode: SaveMode,
+                              parameters: Map[String, String], data: DataFrame): BaseRelation = {
     val relation = new RedisSourceRelation(sqlContext, parameters, userSpecifiedSchema = None)
-
     mode match {
       case Append => relation.insert(data, overwrite = false)
       case Overwrite => relation.insert(data, overwrite = true)
       case ErrorIfExists =>
-        // TODO: check if exists
+        if (relation.nonEmpty) {
+          throw new IllegalStateException("Relation is not empty")
```

Comment:
I think this message may not be clear for the user. Please change to something like 'SaveMode is set to ErrorIfExists and dataframe already exists in Redis and contains data.'